src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -128,33 +133,29 @@ public Result getFirstResult() {
         return null;
     }
 
-    // TODO NR, SE: remove redundant setter after relationship change on client. Currently we need two deserializing setters for "result" (client) and "results" (server)
-    @JsonProperty(value = "result", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResult(Result result) {
-        this.setResults(result);
-    }
-
-    // TODO Ruscher, Entholzer: refactor to addResult
     /**
-     * custom setter that supports the migration from 1...1 to 1...* in the submission->result(s) relationship
-     * Will be refactore in the future
-     * @param result the result that should be added, in case this is null, an empty list will be used instead
+     * Used as a setResult method, as typically the latest result is used
+     *
+     * @param result
      */
-    @JsonProperty(value = "results", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResults(Result result) {
-        if (result == null) {
-            // clear the list of results
+    // @JsonProperty(value = "result", access = JsonProperty.Access.WRITE_ONLY)
+    public void replaceLatestOrIfEmptyAddResult(Result result) {
+        if (this.results == null) {
             this.results = new ArrayList<>();
         }
+        else if (!this.results.isEmpty()) {
+            results.set(results.size() - 1, result);
```

Comment:
I am not quire sure what is happening here. As far as I know we only have one result at the moment, therefore this would remove the last result (the one result at the moment) and then add the same result back, right? Or am I misunderstanding here something?

Reply:
It removes the last result, but the result passed to the method isn't the same result. So, a different result is set to the results list.